woocommerce/woocommerce
# Add new E2E tests to cover Shopper Checkout Block flow

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes milestone [#32](https://github.com/woocommerce/woocommerce-quality/milestone/32) in Woo Quality repo

Added tests to cover new checkout experience (block-based).

- `checkout-block.spec.js`
- `checkout-block-coupons.js`
- `cart-checkout-block-calculate-tax.spec.js`
and improved `cart-block-coupons.spec.js`



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally
2. Execute tests one by one: 
- `pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/checkout-block.spec.js`
- `pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/checkout-block-coupons.spec.js`
- `pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/cart-checkout-block-calculate-tax.spec.js`
- `pnpm test:e2e-pw ./tests/e2e-pw/tests/shopper/cart-block-coupons.spec.js`
4. Make sure the CI passes



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
